src/tasks/installers/operator.ts

```diff
@@ -60,6 +61,68 @@ export class OperatorTasks {
           }
         }
       },
+      {
+        title: 'Checking for pre-created TLS secret',
+        // In case of Openshift infrastructure the certificate from cluster router will be used, so no need in the `che-tls` secret.
+        skip: () => !isKubernetesPlatformFamily(flags.platform),
+        task: async (_: any, task: any) => {
+          // Che is being deployed on Kubernetes infrastructure
+
+          if (! await this.checkTlsMode(flags)) {
+            // No TLS mode, skip this check
+            return
+          }
+
+          const cheSecretName = 'che-tls'
+          const cheSecret = await kube.getSecret(cheSecretName, flags.chenamespace)
+          if (cheSecret) {
+            task.title = `${task.title}... "${cheSecretName}" secret found`
+            return
+          }
+
+          // The secret is required but doesn't exist, show error message.
+          const errorMessage =
+            `Che TLS mode is turned on, but required "${cheSecretName}" secret is not pre-created in "${flags.chenamespace}" namespace, so Eclipse Che cannot be started. \n` +
+            'This is not bug in Eclipse Che and such behavior is expected. \n' +
+            'Please refer to Che documentation for more informations: ' +
+            'https://www.eclipse.org/che/docs/che-7/setup-che-in-tls-mode-with-self-signed-certificate/'
+          throw new Error(errorMessage)
+        }
+      },
+      {
+        title: 'Checking certificate',
+        // If the flag is set no need to check if it is required
+        skip: () => flags['self-signed-cert'],
+        task: async (_: any, task: any) => {
+          if (! await this.checkTlsMode(flags)) {
+            // No TLS mode, skip this check
+            return
+          }
+
+          const warningMessage = 'Self-signed certificate is used, so "--self-signed-cert" option is required. Added automatically.'
+
+          const platform = flags.platform
+          if (platform === 'minikube' || platform === 'crc' || platform === 'minishift') {
+            // There is no way to use real certificate on listed above platforms
+            cli.warn(warningMessage)
+            flags['self-signed-cert'] = true
+            task.title = `${task.title}... self-signed`
+            return
+          }
+
+          if (flags.domain && (flags.domain.endsWith('nip.io') || flags.domain.endsWith('xip.io'))) {
+            // It is not possible to use real certificate with *.nip.io and similar services
+            cli.warn(warningMessage)
+            flags['self-signed-cert'] = true
+            task.title = `${task.title}... self-signed`
+            return
+          }
+
+          // TODO check the secret certificate if it is commonly trusted.
+          cli.info('TLS mode is turned on, however we failed to determine whether self-signed certificate is used. \n\
```

Comment:
warn?

Reply:
I used info here because it is more like a message. Think of case when real certificate is used. Then we would end up in a situation with unresolvable warning, which is wrong in my opinion.